src/Entities/Player.cpp

```diff
@@ -462,7 +462,17 @@ void cPlayer::SetTouchGround(bool a_bTouchGround)
 	static const auto HalfWidth = GetWidth() / 2;
 	static const auto EPS = 0.0001;
 
-	BLOCKTYPE BlockAtFoot = GetWorld()->GetBlock(POS_TOINT);
+	/* Since swimming is decided in a tick and is asynchronous to this, we have to check for dampeners ourselves.
```

Comment:
I might have misread. Either way even in the same thread the swimming state is set AFTER the damage is dealt, which won't work.